tests/sets/t15435.nim

```diff
@@ -0,0 +1,29 @@
+discard """
+errormsg: "type mismatch: got <set[uint8], set[range 1..5(uint8)]>"
+nimout: '''t15435.nim(28, 13) Error: type mismatch: got <set[uint8], set[range 1..5(uint8)]>
+but expected one of:
+proc `<`[T](x, y: set[T]): bool
+  first type mismatch at position: 2
+  required type for y: set[T]
+  but expression 'x' is of type: set[range 1..5(uint8)]
+20 other mismatching symbols have been suppressed; compile with --showAllMismatches:on to see them
+
+expression: {1'u8, 5} < x'''
+"""
+
+
+
+
+
+
+
```

Comment:
`# bug #15435`

specify the issues related for search convenience.

Reply:
> specify the issues related for search convenience.

I just added a commit to do this. But this test still needs to be separate, right? Because this test is testing a compile fail and I cannot make it a part of `tsets.nim`. Btw, before I created t15435.nim, I took a look at tsets.nim, but it has this at the top:

> # xxx these tests are not very good, this should be revisited.